File: 卡码/代码随想录/Go/贪心/02-中等/LeetCode_0376.go
//376. 摆动序列
package main

func wiggleMaxLength(nums []int) int {
	if len(nums)<=1 {
		return len(nums)
	}
	res,pre,cur:=1,0,0
	for i:=1;i<len(nums);i++ {
		cur=nums[i]-nums[i-1]
		if cur>0&&pre<=0||cur<0&&pre>=0 {
			res++
			pre=cur
		}
	}
	return res
}

File: 卡码/代码随想录/Python/贪心/02-中等/LeetCode_0376.py
#376. 摆动序列
from typing import *

class Solution:
  def wiggleMaxLength(self, nums: List[int]) -> int:
    if len(nums)<=1:
      return len(nums)
    res:int=1
    pre:int=0
    cur:int=0
    for i in range(1,len(nums)):
      cur=nums[i]-nums[i-1]
      if cur>0 and pre<=0 or cur<0 and pre>=0:
        pre=cur
        res+=1
    return res